### ISSUE TITLE
Option to disable `ruby-lsp-rails` server-dependent addon features [low-prio]

### ISSUE THREAD

**Issue by Splines** [Shopify/ruby-lsp#2038]
### I have checked that this feature is not already implemented

- [X] This feature does not exist

### Use case

We employ a Docker setup for our e-learning plattform [MaMpf](https://github.com/MaMpf-HD/mampf). We make use of the awesome Ruby LSP VSCode extension.

In our logs (Output pane of VSCode), we find that `Server dependent features will not be available`. This is reasonable due to [these lines](https://github.com/Shopify/ruby-lsp-rails/blob/23edfb1e670494159bfbf0f6316deaa74266372a/lib/ruby_lsp/ruby_lsp_rails/runner_client.rb#L24-L28) in the [`ruby-lsp-rails` addon](https://github.com/Shopify/ruby-lsp-rails) since a connection to our database cannot be established as the database is running inside a Docker container, while the Ruby LSP extension only accesses our local files (in a WSL environment).

### Description

As we don't need the server-dependent features of Ruby LSP (*), we propose introducing an option to disable automatic bundling/activation of the `ruby-lsp-rails` addon/gem such that Ruby LSP does not try to communicate/instantiate our Rails app and instead only relies on static analysis of the code at hand.

<sub>(*) Of course, these additional features would be nice to have, but the static analysis is sufficient for us. Or rather, spawning a new Docker container every time one saves a Ruby file would be way too slow. And the setup would evolve quite some effort to get the Ruby LSP commands routed to the Docker container correctly, that we don't see the benefits for right now. And note that development inside a docker container as described [in your VSCode extension Readme (section title "Developing on containers")](https://marketplace.visualstudio.com/items?itemName=Shopify.ruby-lsp) is not an option for us.</sub>

### Implementation

Include a check in [`setup_bundler.rb`](https://github.com/Shopify/ruby-lsp/blob/main/lib/ruby_lsp/setup_bundler.rb) and make it available as flag in the VSCode settings.

For us, this issue is of _low priority_, as it's just a warning/error we don't want to see in the logs, but apart from that, everything works fine for us.

**Comment by Splines** [comment]
Yes, we can indeed run any rails commands (e.g. `rails console`) inside our "main" docker container using `docker compose exec -it` (or `docker compose run -it` to start a _new_ container for interactively executing the specified command).

For RSpec tests, we use the [Ruby Test Explorer extension](https://marketplace.visualstudio.com/items?itemName=connorshea.vscode-ruby-test-adapter) and specify in its settings that it should execute [a custom python script](https://github.com/MaMpf-HD/mampf/blob/3d1e31939eef39512d2b1ba629c9239fc083e419/spec/rspec_inside_docker.py#L59) when running the tests to make sure they are executed inside a new docker test container. While the execution of the tests itself is quite fast, **the startup of the container takes quite a few seconds** (haven't measured, but my guess is around 15 to 20s on my i7 4 core, 16 GB RAM machine and that feels like an eternity).

That's just why I fear that even with a customized Rails runner command path for the Ruby LSP extension, we'd rather not use a "dockerized" command as I don't want to wait 15s every time I change something in a Ruby file in VSCode in order to be able to use the features of Ruby LSP. Hence, I think I'd prefer an option to only use static language features of the language server and deactivate the "dynamic" ones that rely on executing actual Ruby commands against the project. <sub>Not using Docker at all is not an option for us as it's just too much of a hassle to set up every component of our whole infrastructure individually (e.g. database, mail server and so forth...).</sub>

**Comment by Splines** [comment]
Unfortunately, it fails due to the [condition](https://github.com/Shopify/ruby-lsp-rails/blob/a48b2a6a095e3f4adc03608b04d500c54d198c41/lib/ruby_lsp/ruby_lsp_rails/server.rb#L62-L63) `ActionDispatch::Routing::Mapper.respond_to?(:route_source_locations) && ActionDispatch::Routing::Mapper.route_source_locations` not being satisfied. And yes, the [line you mentioned](https://github.com/Shopify/ruby-lsp-rails/blob/23edfb1e670494159bfbf0f6316deaa74266372a/lib/ruby_lsp/ruby_lsp_rails/server.rb#L76) would probably return an incorrect path that one would have to translate back to the local filesystem path.

I have to be honest and say that the features I deem relevant for a nice development experience already come shipped with Ruby LSP, even without `ruby-lsp-rails`, i.e. getting semantic highlighting, jumping to definitions, seeing an outline, getting documentation links etc. While the extra features provided by Ruby LSP Rails would be nice to have, personally I can live without them. Having setup our app in Docker containers, it's of course more effort to get Ruby LSP Rails to work nicely if communication has to happen with the Rails app.

> We previously did offer a way to disable the server addon features via an initializer but we deprecated it when we changed the server approach in https://github.com/Shopify/ruby-lsp-rails/pull/256. We can considered making that option available again if necessary.

I feel like that'd be the preferred way for us based on my first little experiments here. If you could make this option available again somehow, it'd be much appreciated. For now, we will continue to ignore the `Server dependent features will not be available` message and the accompanying error in the logs.

**Comment by andyw8** [comment]
(I'll move this to the ruby-lsp-rails repo since that's where the change would be needed)
